eth/api_migration.go

```diff
@@ -0,0 +1,352 @@
+			return result, err
+		}
+	}
+
+	// For hash-based queries or recent blocks, try local first, if not found attempt to proxy to erigon
+	result, err := api.BlockChainAPI.GetStorageAt(ctx, address, hexKey, blockNrOrHash)
+	if err == nil && result != nil {
+		return result, nil
+	}
+
+	if api.config != nil && api.config.ErigonClient != nil {
+		var remoteResult hexutil.Bytes
+		err := api.config.ErigonClient.CallContext(ctx, &remoteResult, "eth_getStorageAt", address, hexKey, blockNrOrHash)
+		return remoteResult, err
+	}
+
+	return result, err
+}
+
+// MigrationTransactionAPI wraps the standard TransactionAPI to add migration routing
+type MigrationTransactionAPI struct {
+	*ethapi.TransactionAPI
+	config *MigrationConfig
+}
+
+// NewMigrationTransactionAPI creates a new migration-aware TransactionAPI
+func NewMigrationTransactionAPI(original *ethapi.TransactionAPI, config *MigrationConfig) *MigrationTransactionAPI {
+	return &MigrationTransactionAPI{
+		TransactionAPI: original,
+		config:         config,
+	}
+}
+
+// GetTransactionByHash returns the transaction for the given hash
+func (api *MigrationTransactionAPI) GetTransactionByHash(ctx context.Context, hash common.Hash) (*ethapi.RPCTransaction, error) {
+	// Try local first
+	tx, err := api.TransactionAPI.GetTransactionByHash(ctx, hash)
+	if err == nil && tx != nil {
+		return tx, nil
+	}
+
+	// If not found locally and migration is configured, try erigon
+	if api.config != nil && api.config.ErigonClient != nil {
+		var result *ethapi.RPCTransaction
+		err := api.config.ErigonClient.CallContext(ctx, &result, "eth_getTransactionByHash", hash)
+		if err == nil && result != nil {
+			return result, nil
+		}
+	}
+
+	return tx, err
+}
+
+// GetTransactionReceipt returns the receipt for the given transaction hash
+func (api *MigrationTransactionAPI) GetTransactionReceipt(ctx context.Context, hash common.Hash) (map[string]interface{}, error) {
+	// Try local first
+	receipt, err := api.TransactionAPI.GetTransactionReceipt(ctx, hash)
+	if err == nil && receipt != nil {
+		return receipt, nil
+	}
+
+	// If not found locally and migration is configured, try erigon
+	if api.config != nil && api.config.ErigonClient != nil {
+		var result map[string]interface{}
+		err := api.config.ErigonClient.CallContext(ctx, &result, "eth_getTransactionReceipt", hash)
+		if err == nil && result != nil {
+			return result, nil
+		}
+	}
+
+	return receipt, err
+}
+
+type MigrationFilterAPI struct {
+	*filters.FilterAPI
+	config *MigrationConfig
+}
+
+// NewMigrationFilterAPI creates a new migration-aware FilterAPI
+func NewMigrationFilterAPI(original *filters.FilterAPI, config *MigrationConfig) *MigrationFilterAPI {
+	return &MigrationFilterAPI{
+		FilterAPI: original,
+		config:    config,
+	}
+}
+
+func (api *MigrationFilterAPI) GetLogs(ctx context.Context, crit filters.FilterCriteria) ([]*types.Log, error) {
```

Comment:
[eth_newFilter](https://ethereum.org/zh/developers/docs/apis/json-rpc/#eth_newfilter) method require to handle: